In [1]:
import sys
from pathlib import Path
import os
import json

sys.path.append(str(Path.cwd().parent.parent))
print(str(Path.cwd().parent.parent))
print("Current working directory: ", os.getcwd())

from ModelDevelopment.CentralPipeline import CentralPipeline
from ModelDevelopment.ImageBatchPipeline import ImageBatchPipeline
from DataProcessing.DataPreProcessing import DataPaths

c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition
Current working directory:  c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\ModelDevelopment\experiments


In [2]:
# gt_test = DataPaths.CHALLENGE_DATA_GT.value
# #Schema of gt_test: {"0": 4, "1": 93, "2": 25, "3": -1, "4": -1,

# # load the json and compile a list of all tracklets with -1 as not_legible and all other tracklets as legible
# with open(gt_test, 'r') as f:
#     data = json.load(f)
#     legible = []
#     illegible = []
    
#     for key, value in data.items():
#         if value == -1:
#             illegible.append(key)
#         else:
#             legible.append(key)
    
# print("Legible tracklets: ", legible)
# print("Illegible tracklets: ", illegible)

In [3]:
# # Delete soccer_ball.json files
# import os
# root_path = r"C:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\test\images"

# for tracklet in range(1, 1210):
#   path = os.path.join(root_path, str(tracklet))
  
#   # If there is a .soccer_ball.json file in there, delete it
#   if os.path.exists(os.path.join(path, "soccer_ball.json")):
#     os.remove(os.path.join(path, "soccer_ball.json"))

In [4]:
tracklets_to_process_override_batch1 = [i for i in range(0, 605)] # COLIN
tracklets_to_process_override_batch2 = [i for i in range(605, 1210)] # JARED Second half of the tracklets (605 items)

pipeline = CentralPipeline(
  #tracklets_to_process_override=[109, 88, 34, 151, 19, 68, 67, 128, 57, 178, 189, 196, 214, 242, 258, 271, 304, 327, 320, 352, 375, 378, 414, 416, 439, 425, 451, 453, 454, 466, 499, 511, 513, 526, 528, 567, 576, 594, 619, 623, 646, 687, 682, 709, 711, 717, 739, 758, 767, 795, 788, 840, 843, 832, 841, 852, 847, 851, 856, 850, 868, 866, 871, 901, 902, 904, 900, 907, 928, 939, 944, 961, 984, 983, 1005, 1039, 1064, 1063, 1077, 1091, 1101, 1137, 1152, 1153, 1160, 1167, 1157, 1165, 1206],
  #tracklets_to_process_override=[327, 711, 907, 1064], # These ones are now passing
  #tracklets_to_process_override=[327]
  #tracklets_to_process_override=[0],
  #tracklets_to_process_override=illegible,
  #tracklets_to_process_override=[168, 167, 162, 97, 39, 87, 89, 134, 61],
  #tracklets_to_process_override=[168, 167, 162, 97, 39, 87, 89, 134, 61, 172, 164, 199, 215, 260, 267, 269, 281, 276, 288, 295, 317, 325, 340, 345, 374, 394, 388, 386, 391, 406, 408, 434, 437, 459, 471, 514, 532, 558, 683, 748, 770, 790, 802, 827, 861, 869, 879, 897, 895, 909, 964, 981, 999, 1023, 1031, 1025, 1033, 1030, 1055, 1088, 1087, 1169, 1175, 1174, 1173, 1183, 1170, 1179, 1184, 1185],
  #num_tracklets=1210,
  tracklets_to_process_override=tracklets_to_process_override_batch1,
  #tracklets_to_process_override=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
  #num_images_per_tracklet=50,
  input_data_path=DataPaths.TEST_DATA_DIR.value,
  output_processed_data_path=DataPaths.PROCESSED_DATA_OUTPUT_DIR_TEST.value,
  common_processed_data_dir=DataPaths.COMMON_PROCESSED_OUTPUT_DATA_TEST.value,
  gt_data_path=DataPaths.TEST_DATA_GT.value,
  display_transformed_image_sample=False, # NOTE: DO NOT USE. Code is parallelized so we cannot show images anymore. Code breaks, but first one will show if True.
  use_cache=True, # Set to false to rebuild the cache
  suppress_logging=False,
  use_image_enhancement=True,
  
  # --- PARALLELIZATION PARAMS --- These settings are optimal for an NVIDIA RTX 3070 Ti Laptop GPU.
  num_workers=2,            # CRITICAL optimisation param. Adjust accordingly. Pre-processing: 4 | STR: 2 | Crops: 4
  tracklet_batch_size=32,   # CRITICAL optimisation param. Adjust accordingly. Always 32
  image_batch_size=100,     # CRITICAL optimisation param. Adjust accordingly. Pre-processing: 200 | STR: 100 | Crops: 200
  num_threads_multiplier=5  # CRITICAL optimisation param. Adjust accordingly. Pre-processing: 4
  )                         #                              Interpretation: num_threads = num_workers * num_threads_multiplier

2025-03-31 09:17:12 [INFO] DataPreProcessing initialized. Universe of available data paths:
2025-03-31 09:17:12 [INFO] ROOT_DATA_DIR: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted
2025-03-31 09:17:12 [INFO] TEST_DATA_GT: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\test\test_gt.json
2025-03-31 09:17:12 [INFO] TRAIN_DATA_GT: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\train\train_gt.json
2025-03-31 09:17:12 [INFO] TEST_DATA_DIR: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\test\images
2025-03-31 09:17:12 [INFO] TRAIN_DATA_DIR: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\train\images
2025-03-31 09:17:12 [INFO] CHALLENGE_DATA_DIR: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\challenge\images
2

In [ ]:
pipeline.run_soccernet(
  run_soccer_ball_filter=False,
  generate_features=True,
  run_filter=True,
  run_legible=True,
  run_legible_eval=True,
  run_pose=True,
  run_crops=True,
  run_str=True,
  run_combine=False,
  run_eval=False,
  use_clip4str=True,
  pyscript=False,
  )

2025-03-31 09:17:13 [INFO] Running the SoccerNet pipeline.
2025-03-31 09:17:13 [INFO] Tracklet override applied. Using provided tracklets: 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 1

  0%|          | 0/19 [00:00<?, ?it/s]

2025-03-31 09:17:13 [INFO] Tracklet batch size: 32
2025-03-31 09:17:13 [INFO] Image batch size: 100
2025-03-31 09:17:13 [INFO] Number of workers: 2
2025-03-31 09:17:13 [INFO] Number of threads created: 10
2025-03-31 09:17:13 [INFO] Using double parallelization: multiprocessing + CUDA batch processing.


Processing tracklets (CUDA + CPU):   0%|          | 0/32 [00:00<?, ?it/s]

2025-03-31 09:17:48 [INFO] Creating placeholder data files for Soccer Ball Filter.


Processing Batch Tracklets (0-32):   0%|          | 0/32 [00:00<?, ?it/s]

c:\Users\colin\miniconda3\envs\UBC\Lib\site-packages\pytorch_lightning\utilities\migration\migration.py:208: You have multiple `ModelCheckpoint` callback states in this checkpoint, but we found state keys that would end up colliding with each other after an upgrade, which means we can't differentiate which of your checkpoint callbacks needs which states. At least one of your `ModelCheckpoint` callbacks will not be able to reload the state.
Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 09:17:51 [INFO] Saved features for tracklet with shape (354, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\1\features.npy
2025-03-31 09:17:51 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 09:17:53 [INFO] Saved features for tracklet with shape (461, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\3\features.npy
2025-03-31 09:17:53 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 09:17:55 [INFO] Saved features for tracklet with shape (311, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\4\features.npy
2025-03-31 09:17:55 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 09:17:57 [INFO] Saved features for tracklet with shape (490, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\6\features.npy
2025-03-31 09:17:57 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 09:18:00 [INFO] Saved features for tracklet with shape (544, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\8\features.npy
2025-03-31 09:18:00 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 09:18:02 [INFO] Saved features for tracklet with shape (465, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\0\features.npy
2025-03-31 09:18:02 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 09:18:03 [INFO] 2025-03-31 09:18:02 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 09:18:02 [INFO] Round 1: Excluded 4 outliers
2025-03-31 09:18:02 [INFO] Round 1: Average score: 0.9969872832298279
2025-03-31 09:18:02 [INFO] Excluded images: ['1_196.jpg', '1_197.jpg', '1_594.jpg', '1_595.jpg']

2025-03-31 09:18:03 [ERROR] 
2025-03-31 09:18:03 [INFO] Done removing outliers
2025-03-31 09:18:03 [INFO] Running model chain on preprocessed image(s).
2025-03-31 09:18:03 [INFO] Classifying legibility of image(s) using pre-trained model.


c:\Users\colin\miniconda3\envs\UBC\Lib\site-packages\torchvision\models\_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
c:\Users\colin\miniconda3\envs\UBC\Lib\site-packages\torchvision\models\_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=ResNet50_Weights.IMAGENET1K_V1`. You can also use `weights=ResNet50_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)


2025-03-31 09:18:05 [INFO] 2025-03-31 09:18:04 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 09:18:04 [INFO] Round 1: Excluded 6 outliers
2025-03-31 09:18:04 [INFO] Round 1: Average score: 0.9956437349319458
2025-03-31 09:18:04 [INFO] Excluded images: ['3_205.jpg', '3_206.jpg', '3_267.jpg', '3_379.jpg', '3_380.jpg', '3_611.jpg']

2025-03-31 09:18:05 [ERROR] 
2025-03-31 09:18:05 [INFO] Done removing outliers
2025-03-31 09:18:05 [INFO] Running model chain on preprocessed image(s).
2025-03-31 09:18:05 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 09:18:06 [INFO] Saved features for tracklet with shape (627, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\5\features.npy
2025-03-31 09:18:06 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 09:18:09 [INFO] 2025-03-31 09:18:08 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 09:18:08 [INFO] Round 1: Excluded 5 outliers
2025-03-31 09:18:08 [INFO] Round 1: Average score: 0.9941612482070923
2025-03-31 09:18:08 [INFO] Excluded images: ['4_199.jpg', '4_200.jpg', '4_387.jpg', '4_388.jpg', '4_389.jpg']

2025-03-31 09:18:09 [ERROR] 
2025-03-31 09:18:09 [INFO] Done removing outliers
2025-03-31 09:18:09 [INFO] Running model chain on preprocessed image(s).
2025-03-31 09:18:09 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 09:18:11 [INFO] Saved features for tracklet with shape (595, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\9\features.npy
2025-03-31 09:18:11 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 09:18:13 [INFO] 2025-03-31 09:18:12 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 09:18:12 [INFO] Round 1: Excluded 2 outliers
2025-03-31 09:18:12 [INFO] Round 1: Average score: 0.9982720613479614
2025-03-31 09:18:12 [INFO] Excluded images: ['6_489.jpg', '6_490.jpg']

2025-03-31 09:18:13 [ERROR] 
2025-03-31 09:18:13 [INFO] Done removing outliers
2025-03-31 09:18:13 [INFO] Running model chain on preprocessed image(s).
2025-03-31 09:18:13 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 09:18:17 [INFO] Saved features for tracklet with shape (661, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\7\features.npy
2025-03-31 09:18:17 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 09:18:18 [INFO] 2025-03-31 09:18:16 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 09:18:16 [INFO] Round 1: Excluded 7 outliers
2025-03-31 09:18:16 [INFO] Round 1: Average score: 0.9966508150100708
2025-03-31 09:18:16 [INFO] Excluded images: ['8_402.jpg', '8_403.jpg', '8_611.jpg', '8_612.jpg', '8_613.jpg', '8_616.jpg', '8_621.jpg']

2025-03-31 09:18:18 [ERROR] 
2025-03-31 09:18:18 [INFO] Done removing outliers
2025-03-31 09:18:18 [INFO] Running model chain on preprocessed image(s).
2025-03-31 09:18:18 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 09:18:22 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.47921140491962433, 0.47122254473822456, 0.49301019310951233
2025-03-31 09:18:22 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.0, 0.0, 0.0
2025-03-31 09:18:22 [INFO] Tracklet 1 is illegible.
2025-03-31 09:18:22 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recogniti

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 09:18:24 [INFO] 2025-03-31 09:18:22 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 09:18:22 [INFO] Round 1: Excluded 7 outliers
2025-03-31 09:18:22 [INFO] Round 1: Average score: 0.9971559047698975
2025-03-31 09:18:22 [INFO] Excluded images: ['0_176.jpg', '0_281.jpg', '0_282.jpg', '0_350.jpg', '0_526.jpg', '0_649.jpg', '0_650.jpg']

2025-03-31 09:18:24 [ERROR] 
2025-03-31 09:18:24 [INFO] Done removing outliers
2025-03-31 09:18:24 [INFO] Running model chain on preprocessed image(s).
2025-03-31 09:18:24 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 09:18:29 [INFO] Saved features for tracklet with shape (549, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\10\features.npy
2025-03-31 09:18:30 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-31 09:18:30 [INFO] 2025-03-31 09:18:29 [INFO] Threshold for scores: 0.97, rounds: 1
202

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 09:18:36 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5137666165828705, 0.5125175140846788, 0.5334370732307434
2025-03-31 09:18:36 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.514610230922699, 0.5148228122128381, 0.5334370732307434
2025-03-31 09:18:36 [INFO] Tracklet 4 is legible.
2025-03-31 09:18:36 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\4\legible_results.json
2025-03-31 09:18:36 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\4\legible_results.json
2025-03-31 09:18:36 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\4\illegible_results.json
2025-03-31 09:18:36 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognit

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


## Benchmark (current pipeline metrics)
### Results
- 2025-03-29 14:44:05 [INFO] Correct 1051 out of 1210. Accuracy 86.85950413223141%.
- 2025-03-29 14:44:05 [INFO] TP=766, TN=285, FP=70, FN=89
- 2025-03-29 14:44:05 [INFO] Precision=0.916267942583732, Recall=0.895906432748538
- 2025-03-29 14:44:05 [INFO] F1=0.9059727971614429
- 2025-03-29 14:44:05 [INFO] Saved failed legibility cases to:  c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\common_data\failed_legibility_cases.json

### Analysis
legible_but_marked_illegible: 89 cases
109, 88, 34, 151, 19, 68, 67, 128, 57, 178, 189, 196, 214, 242, 258, 271, 304, 327, 320, 352, 375, 378, 414, 416, 439, 425, 451, 453, 454, 466, 499, 511, 513, 526, 528, 567, 576, 594, 619, 623, 646, 687, 682, 709, 711, 717, 739, 758, 767, 795, 788, 840, 843, 832, 841, 852, 847, 851, 856, 850, 868, 866, 871, 901, 902, 904, 900, 907, 928, 939, 944, 961, 984, 983, 1005, 1039, 1064, 1063, 1077, 1091, 1101, 1137, 1152, 1153, 1160, 1167, 1157, 1165, 1206

illegible_but_marked_legible: 70 cases
168, 167, 162, 97, 39, 87, 89, 134, 61, 172, 164, 199, 215, 260, 267, 269, 281, 276, 288, 295, 317, 325, 340, 345, 374, 394, 388, 386, 391, 406, 408, 434, 437, 459, 471, 514, 532, 558, 683, 748, 770, 790, 802, 827, 861, 869, 879, 897, 895, 909, 964, 981, 999, 1023, 1031, 1025, 1033, 1030, 1055, 1088, 1087, 1169, 1175, 1174, 1173, 1183, 1170, 1179, 1184, 1185

- Total misclassifications: 159
- False negatives (legible marked as illegible): 89
- False positives (illegible marked as legible): 70
- Percentage of false negatives: 7.36%
- Percentage of false positives: 5.79%
- Combined percentage of errors as a percentage of total cases: 13.14%

In [ ]:
import json
failed_cases = f"{DataPaths.COMMON_PROCESSED_OUTPUT_DATA_CHALLENGE.value}/failed_legibility_cases.json"

# Load this JSON file
with open(failed_cases, 'r') as f:
    failed_cases_dict = json.load(f)
    # Print the every key and its associated value (value will be a list so need to make that printable with ",".join):
    for key, value in failed_cases_dict.items():
        print(f"{key}: {len(value)} cases")
        if len(value) > 0:
            print(f"{', '.join(value)}")
        print()
    
    # Summary statistics
    legible_but_marked_illegible = len(failed_cases_dict.get("legible_but_marked_illegible", []))
    illegible_but_marked_legible = len(failed_cases_dict.get("illegible_but_marked_legible", []))
    total_errors = legible_but_marked_illegible + illegible_but_marked_legible
    
    print(f"Total misclassifications: {total_errors}")
    print(f"False negatives (legible marked as illegible): {legible_but_marked_illegible}")
    print(f"False positives (illegible marked as legible): {illegible_but_marked_legible}")
    print(f"Percentage of false negatives: {legible_but_marked_illegible / 1210 * 100:.2f}%")
    print(f"Percentage of false positives: {illegible_but_marked_legible / 1210 * 100:.2f}%")
    print(f"Combined percentage of errors as a percentage of total cases: {total_errors / 1210 * 100:.2f}%")

In [ ]:
# import timm
# import torch
# from torch import nn

# model = timm.create_model("timm/vit_base_patch16_224.orig_in21k_ft_in1k", pretrained=False)
# model.head = nn.Linear(model.head.in_features, 10)
# model.load_state_dict(
#     torch.hub.load_state_dict_from_url(
#         "https://huggingface.co/edadaltocg/vit_base_patch16_224_in21k_ft_svhn/resolve/main/pytorch_model.bin",
#         map_location="cpu",
#         file_name="vit_base_patch16_224_in21k_ft_svhn.pth",
#     )
# )